app/models/agents/post_agent.rb

```diff
@@ -122,6 +128,10 @@ def validate_options
         errors.add(:base, "if provided, no_merge must be 'true' or 'false'")
       end
 
+      if options['output_mode'].present? && !%[clean merge].include?(options['output_mode'].to_s)
```

Comment:
You are using `interpolated['output_mode']` below which means the user could use liquid to switch the mode based on the incoming event. If that is intended we should skip the validation if `options['output_mode'].include?('{{')` is true.